### ISSUE TITLE
Lex the token after `as` as a name

### ISSUE THREAD

**Issue by tconbeer** [tconbeer/sqlfmt#266]
**Describe the bug**
We are not going to create a full SQL grammar, but we should know that tokens that appear after the `as` keyword are aliases, and therefore are ordinary names. This should help safeguard against side affects when adding an even larger list of keywords to the polyglot dialect (at least a little bit). (This won't help with aliases that don't use `as` but shame on those people).

**To Reproduce**
After #bac612a, we recognize `values` as an unterminated keyword. But the http_archive project uses it as an alias for both tables and fields in a few files, and we do bad things to those files. [Link](https://github.com/tconbeer/http_archive_almanac/pull/9/files)

**Expected behavior**
```sql
from 
    `httparchive.almanac.parsed_css`, 
    unnest(getfontvariationsettings(css)) as values, 
    unnest(split(values, ',')) as value
```

**Actual behavior**
```sql
from `httparchive.almanac.parsed_css`, unnest(getfontvariationsettings(css)) as
values, unnest(split(values, ',')) as value
```

**Additional context**
What is the output of `sqlfmt --version`?
#bac612a, v0.12.0 pre

**Comment by tconbeer** [comment]
`as` can also be followed by parens, (e.g., ctes)

**Comment by owenprough-sift** [comment]
> but we should know that tokens that appear after the `as` keyword are aliases, and therefore are ordinary names
Counter example: `cast( col_a as varchar )`. In this example, `varchar` is not an alias but is a keyword.

**Comment by tconbeer** [comment]
Yep! Have to be careful with other types of statements (create function/table) also
